src/id_mapper/graph.py

```diff
@@ -13,18 +13,37 @@
 # limitations under the License.
 
 import logging
+import os
+import time
 
-from py2neo import Node, Relationship
+from py2neo import Graph, Node, Relationship
+from py2neo.packages.httpstream.http import SocketError
 
 
 logger = logging.getLogger(__name__)
 
+connected = False
+while not connected:
+    try:
+        graph = Graph(
+            os.environ['ID_MAPPER_API'],
+            http_port=int(os.environ['ID_MAPPER_PORT']),
+            user=os.environ['ID_MAPPER_USER'],
+            password=os.environ['ID_MAPPER_PASSWORD'],
+        )
+        logger.info("Connected to graph db")
+        connected = True
+    except SocketError as error:
+        logger.info(f"{str(error)}: Could not connect to database, retrying in "
+                    "5 seconds...")
+        time.sleep(5)
+
 
 class Is(Relationship):
     pass
 
 
-def insert_pairs(graph, label, pair1, pair2, organism=None):
+def insert_pairs(label, pair1, pair2, organism=None):
     """Merge nodes to database and create mutual IS relationships between
 
     :param graph: Graph
```

Comment:
Perhaps remove this parameter description from the docstring since you've removed it from the function too.